multicamera_acquisition/visualization.py

```diff
@@ -9,51 +9,37 @@
 import time
 import matplotlib.pyplot as plt
 import pandas as pd
-
-def get_latest(queue, timeout=0.1):
-    start_time = time.time()
-    try:
-        item = queue.get(timeout=timeout)
-        while True:
-            try:
-                elapsed_time = time.time() - start_time
-                remaining_time = timeout - elapsed_time
-
-                if remaining_time > 0:
-                    next_item = queue.get(timeout=remaining_time)
-                    item = next_item
-                else:
-                    break
-            except queue.Empty:
-                break
-    except queue.Empty:
-        item = None
-
-    return item
+import queue as sync_queue
+from collections.abc import Iterable
+from numbers import Number
+import glob, os.path
 
 
 class MultiDisplay(mp.Process):
-    def __init__(
-        self,
-        queues,
-        camera_names,
-        display_ranges,
-        display_downsample=4,
-        cameras_per_row=3,
-        display_size=(300, 300),
-    ):
+
+    def __init__(self, queues, config = None):
         super().__init__()
-        self.pipe = None
+
+        # Store params
+        self.config = config
         self.queues = queues
-        self.camera_names = camera_names
-        self.num_cameras = len(camera_names)
-        self.downsample = display_downsample
-        self.cameras_per_row = cameras_per_row
-        self.display_size = display_size
-        self.display_ranges = display_ranges
 
-    def run(self):
-        """Displays an image to a window."""
+        # Set up the config
+        if config is None:
+            self.config = self.default_display_config()
+        else:
+            self.validate_config()
+
+        # break out config into object attrs
```

Comment:
CW and I talked about not doing this unpacking, and instead keeping everything inside the config, just for absolute highest clarity about where params are coming from. However if a param is generated from info in the config (ie `self.num_cameras = len(self.config["camera_names"])`) then obv we need a new attr